docker/Dockerfile

```diff
@@ -1,12 +1,85 @@
-FROM amd64/debian:stable-slim
-ENV GOTIFY_SERVER_PORT="80"
+ARG BUILDKIT_SBOM_SCAN_CONTEXT=true
+# suppress warnung about invalid variable expansion
+ARG GO_VERSION=PLEASE_PROVIDE_GO_VERSION
+ARG DEBIAN=sid-slim
+
+# Hack to normalize platform to match the chosed build image
+# Get the gotify/build image tag
+ARG __TARGETPLATFORM_DASHES=${TARGETPLATFORM/\//-}
+ARG __TARGETPLATFORM_GO_NOTATION=${__TARGETPLATFORM_DASHES/arm\/v7/arm-7}
+
+# --- JS Builder ---
+
+FROM --platform=${BUILDPLATFORM} node:23 AS js-builder
+
+ARG BUILD_JS=0
+ARG NODE_OPTIONS
+
+COPY . /src/gotify
```

Comment:
(optional) I think the cache ability  could be improved. Currently any change will cause a full rebuild of the image, because of the blank copies. E.g. this could be done for the go part:
```
COPY ./go.* /src/gotify/
RUN (cd /src/gotify && go mod download)
```
and this for node
```
COPY ./ui/package.json ./ui/yarn.lock /src/gotify/ui/
RUN [ "$BUILD_JS" = "1" ] && (cd /src/gotify/ui && yarn install)
```
so that the deps downloads are cached as the yarn install is pretty slow.

Reply:
I removed the dependency from JS to Go code but the other way around I think the dependency still exists because go build will package the JS code.

If BUILD_JS=0 the only overhead is pulling the node image once for the whole build, I think it is fine (Dockerfile should really be scriptable...)

If BUILD_JS=1 docker is smart enough to know there is only one JS build needed cross the whole build.